android-components/components/support/remotesettings/src/main/java/mozilla/components/support/remotesettings/RemoteSettingsClient.kt

```diff
@@ -0,0 +1,52 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.remotesettings
+
+import android.util.Log
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import mozilla.appservices.remotesettings.RemoteSettings
+import mozilla.appservices.remotesettings.RemoteSettingsConfig
+import mozilla.appservices.remotesettings.RemoteSettingsException
+import org.json.JSONObject
+
+/**
+ * Helper class to download collections from remote settings in app services.
+ *
+ * @property serverUrl The optional url for the settings server. If not specified, the standard server will be used.
+ * @property bucketName The optional name of the bucket containing the collection on the server.
+ * If not specified, the standard bucket will be used.
+ * @property collectionName The name of the collection for the settings server.
+ */
+class RemoteSettingsClient(
+    private val serverUrl: String? = "https://firefox.settings.services.mozilla.com",
+    var bucketName: String? = "main",
+    var collectionName: String = "",
+) {
+
+    val config = RemoteSettingsConfig(
+        serverUrl = serverUrl,
+        bucketName = bucketName,
+        collectionName = collectionName,
+    )
+
+    /**
+     * Fetches all records for a collection that can be found in the server.
+     **/
+    suspend fun fetchRemoteServerRecords(): JSONObject = withContext(Dispatchers.IO) {
+        try {
+            val client = RemoteSettings(config)
```

Comment:
It's worth noting that we should be calling this with [use](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/use.html) since it is an [interface in the UniFFI UDL definition](https://searchfox.org/mozilla-mobile/source/application-services/components/remote_settings/src/remote_settings.udl#48). [This conversation](https://mozilla.slack.com/archives/C0559DDDPQF/p1694553760906609) has a bit more detail